File: solutions/cpp/collatz-conjecture/1/collatz_conjecture.cpp
#include "collatz_conjecture.h"

#include <stdexcept>

namespace collatz_conjecture {

int steps(int n){
    if (n <= 0){
        throw std::domain_error("n should be a positive number");
    }
    
    int s = 0;
    while (n != 1){
        n = (n % 2 == 0 ? n / 2 : 3 * n + 1);
        s++;
    }
    return s;
}

}  // namespace collatz_conjecture

File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number):
    if type(number) != int or number < 1:
        raise(ValueError("Only positive integers are allowed"))
    
    n = 0
    while number != 1:
        number = number / 2 if number % 2 == 0 else 3 * number + 1
        n += 1
    return n
